omit keyword should reset to context

bcoca opened the pull request:
ensure we use context/inheritance when calculating value,
  using default only when context is unavailable.

  fixes #75692

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
playbook

bcoca committed:
omit keyword should reset to context

  ensure we use context/inheritance when calculating value,
  using default only when context is unavailable.

  fixes #75692

bcoca committed:
no glob, no cry

bcoca committed:
moved inheritance to context

bcoca committed:
yepoet

bcoca committed:
missing attr

bcoca committed:
restoring should also consider context

bcoca committed:
use testhost

bcoca committed:
added alias for become enable

bcoca committed:
use test user

bcoca committed:
name it

bcoca committed:
sing test dir?

bcoca committed:
nope

bcoca committed:
rolespath

bcoca committed:
cause i start as root in tests

felixfontein commented:
This breaks the openssh_cert integration tests in community.crypto for all `devel`-based tests.

felixfontein commented:
Reproducer:
```
- hosts: localhost
  tasks:
    - name: Make sure foo is gone
      file:
        path: foo
        state: absent
    - name: Create foo - should only be changed in first iteration
      copy:
        dest: foo
        content: foo
      check_mode: '{{ omit }}'
      loop:
        - 1
        - 2
```
The second loop iteration shouldn't do anything because the first one didn't use check mode. But for some reason both of them are run in check mode.

bcoca commented:
I get the expected output, w/o checkmode 
```
#> ansible-playbook test.yml

...

changed: [localhost] => (item=1)
ok: [localhost] => (item=2)
```
Since the first iteration creates the file (changed) and the 2nd sees the file already there.

if foo exists, check-mode:
```
ok: [localhost] => (item=1)
ok: [localhost] => (item=2)
```
nothing to do in either case, file already exists

if foo does not exist, check-mode:
```
changed: [localhost] => (item=1)
changed: [localhost] => (item=2)
```
The 2nd executes as 'changed' cause the first didn't actually make the change so 'detection' will see it still requires a change, while this won't reflect the actual execution ... it is expected due to the limitations of check mode and not a bug.
